src/buckets/components/createBucketForm/MeasurementSchemaSection.tsx

```diff
@@ -25,18 +40,187 @@ if (CLOUD) {
     .MeasurementSchemaList
 }
 
-import {downloadTextFile} from 'src/shared/utils/download'
-
+// note:  once you can add schemas in the create mode,
+// make the second arg required
 interface Props {
   measurementSchemaList?: typeof MeasurementSchemaList
-  isEditing?: boolean
+  onUpdateSchemas?: (schemas: any, b?: boolean) => void
+  showSchemaValidation?: boolean
 }
 interface PanelProps {
   measurementSchema: typeof MeasurementSchema
   index?: number
 }
+interface AddingProps {
+  index: number
+  onAddName: (name: string, isValid: boolean, index: number) => void
+  onAddContents: (columns: string, filename: string, index: number) => void
+  onDelete: (index: number) => void
+  filename?: string
+  name?: string
+  showSchemaValidation?: boolean
+}
+
+const AddingPanel: FC<AddingProps> = ({
+  index,
+  name,
+  onAddName,
+  onAddContents,
+  onDelete,
+  filename,
+  showSchemaValidation,
+}) => {
+  const [schemaName, setSchemaName] = useState(name)
+  // innocent until proven guilty.
+  // this way, when the input first shows up, it doesn't error out immediately,
+  // and once the user puts something in then it gets the real value set.
+  let initialValidity = {valid: true, message: null}
+  if (name) {
+    initialValidity = isNameValid(name)
+  }
+  const [schemaNameValidity, setSchemaNameValidity] = useState(initialValidity)
+  const hasFileError = showSchemaValidation && !filename
+  const [fileError, setFileError] = useState(hasFileError)
+  let fileEMessage = null
+  if (hasFileError) {
+    fileEMessage = 'You must upload a file'
+  }
+  const [fileErrorMessage, setFileErrorMessage] = useState(fileEMessage)
+
+  const setErrorState = (hasError, message) => {
+    setFileError(hasError)
+
+    if (!hasError) {
+      message = null
+    }
+
+    setFileErrorMessage(message)
+  }
+
+  // propagating errors up (or throwing new ones),
+  // since the mini file dnd component calls this method,
+  // then calls setError(false).  so; if we call setError(true) then it gets immediately
+  // overridden.  with the propagation, the error state only gets called once properly.
+  const handleUploadFile = (contents: string, fileName: string) => {
+    // keep swapping out the file, cancel out of the dialog, x out the adding line....etc
+
+    // do parsing here;  to check in the correct format:
+    let columns = null
+    if (contents) {
+      // parse them; if kosher; great!  if not, set errors and do not proceed
+      // don't need to wrap this in try/catch since the caller of this function is inside a try/catch
+      columns = JSON.parse(contents)
```

Comment:
it shouldn't be wrapped in a try/catch; because the caller of this function is inside a try/catch (see the comment above it).
we want the caller of this function; the drag and drop component, to catch the error.  the code is SO much simpler that way.

that way, the miniFileDND component does the correct error decorations, and passes up to its parent (the addingPanel) that there is an error, since most of the errors stay within the MiniFileDnd component.